examples/quantum_die.py

```diff
@@ -28,48 +26,47 @@ def qubits_needed(n):
     """
     return int(math.ceil(math.log(n, 2)))
 
+
+qvm = get_qc('f"{qubits_needed(n)}q-qvm")
+
+
 def die_program(n):
     """
     Generate a quantum program to roll a die of n faces.
     """
-    prog = pq.Program()
-    qubits = qubits_needed(n)
+    prog = Program()
+    n_qubits = qubits_needed(n)
+    ro = prog.declare('ro', 'BIT', n_qubits)
     # Hadamard initialize.
-    for q in range(qubits):
+    for q in range(n_qubits):
         prog.inst(H(q))
     # Measure everything.
-    for q in range(qubits):
-        prog.measure(q, [q])
+    for q in range(n_qubits):
+        prog.measure(q, ro[q])
     return prog
 
-def process_result(r):
-    """
-    Convert a list of measurements to a die value.
-    """
-    return reduce(lambda s, x: 2*s + x, r, 0)
-
-BATCH_SIZE = 10
-
-dice = {}
-
-qvm = api.QVMConnection()
-
+    
 def roll_die(n):
     """
     Roll an n-sided quantum die.
     """
-    addresses = list(range(qubits_needed(n)))
-    if not n in dice:
-        dice[n] = die_program(n)
-    die = dice[n]
-    # Generate results and do rejection sampling.
-    while True:
-        results = qvm.run(die, addresses, BATCH_SIZE)
-        for r in results:
-            x = process_result(r) + 1
-            if 0 < x <= n:
-                return x
+    die_compiled = qvm.compile(die_program(n))
+    return qvm.run(die_compiled)
+   
+def process_results(n):
+    """
+    Convert n digit binary result from the QVM to a value on a die. 
+    """
+    raw_results = roll_die(n)[0]
+    print(raw_results)
+    processing_result = 0
+    for each_qubit_measurement in raw_results:
+        processing_result = 2*processing_result + each_qubit_measurement
+    die_value = processing_result + 1
+    return die_value
+             
 
+             
 if __name__ == '__main__':
     number_of_sides = int(input("Please enter number of sides: "))
     while True:
```

Comment:
Suggest removing this line so that we return only a single roll, rather than running forever. That way the basic unit test I wrote will work. (See PR #611). The unit test checks that the script runs and exits without error, but if the script never exits we can't check the return code.

Reply:
I saw this and it seemed so odd to just run forever that I kept thinking I wasn't seeing it correctly. Would definitely agree with this